signac/contrib/job.py

```diff
@@ -24,31 +24,81 @@
 logger = logging.getLogger(__name__)
 
 
-class _sp_save_hook:
-    """Hook to handle job migration when state points are changed.
+class _LoadAndSaveSingleThread:
+    """A context manager for :class:`SyncedCollection` to wrap saving and loading.
 
-    When a job's state point is changed, in addition
-    to the contents of the file being modified this hook
-    calls :meth:`~Job._reset_sp` to rehash the state
-    point, compute a new job id, and move the folder.
+    It's also not obvious how to achieve thread-safety for statepoint
+    modifications within the current framework because when multiple copies of
+    a job (shallow copies owning the same statepoint) exist and one of them is
+    modified, the calls to reset_statepoint will invalidate the per-file locks
+    because the folders are moved. Since statepoint accesses do not need to be
+    thread safe, this context manager simply removes that functionality.
```

Comment:
I make all these commits to fix drafted docstrings and yet I still miss this... *sigh. Yes, I'll rewrite this.

Reply:
Actually I totally forgot that my design for SyncedCollections already accommodates this requirement in a much cleaner way because I made it possible for classes to indicate that they don't support multithreading using a class-leveling boolean. So this class is superfluous (although the comment is still useful and can be moved). Yay for thinking ahead.